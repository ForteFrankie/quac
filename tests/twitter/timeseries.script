#!/bin/bash
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

# Compute keyword time series and related analyses.

. ./environment.sh

cp $QUACBASE/misc/halloween.xls $DATADIR
cd $DATADIR

exec 2>&1


## ngrams-build ##

# We use several partitions so that we can test that
# hash-based time series lookup works.
echo "$ ngrams-build ..."
ngrams-build --notimes --partitions 10 --min-occur 10 --run 1 --clean --jobdir series $TWEETDIR/pre > /dev/null

x ls -R series
echo "$ dump-pickle total.pkl.gz ..."
dump-pickle series/out/total.pkl.gz


## ngrams-search ##

# "candy" should peak near Halloween. (Note that midnight UTC is 8pm EST, so a
# lot of trick-or-treating related traffic will happen on November 1 UTC.)
y "ngrams-search --query 't@ candy' series/out"
y "ngrams-search --detail --query 't@ candy' series/out"

# "sandy" is the hurricane.
y "ngrams-search --query 't@ sandy' series/out"
y "ngrams-search --detail --query 't@ sandy' series/out"

# We include "a" because it's an n-gram that had trouble being sorted (see
# commit 885d3d)
y "ngrams-search --query 't@ a' series/out"

# Make sure zeros are inserted properly: "taco" has no data on October 28
# (beginning), November 1 (middle), and November 4/5 (end).
y "ngrams-search --detail --query 't@ taco' series/out"


## ngrams-correlate ##

# Build the correlations. The sample rate is so tiny because:
#
#     0.01   spritzer 1%
#   * 0.05   5% sample for testing
#   * 0.015  1.5% of tweets are geotagged
#   * 0.5    last day of test sample is incomplete
#   -------
#   = 3.75e-6
#
echo "$ ngrams-correlate ..."
ngrams-correlate --notimes --min-similarity 0.7 --min-ppm 2000 --tw-sample-rate 3.75e-6 --run 1 --clean --jobdir corr series/out halloween.xls > /dev/null

# Results
x ls -R corr
x head -10 corr/out/halloween:halloween.tsv
x head -10 corr/out/halloween:s%40ndy.tsv
