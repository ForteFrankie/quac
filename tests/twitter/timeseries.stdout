$ ngrams-build ...
ngbld INFO     starting
ngbld INFO     input appears to be tweets
ngbld INFO     loading metadata
ngbld INFO     setting up job
ngbld INFO     writing totals
ngbld INFO     running job
ngbld INFO     cleaning up job
ngbld INFO     done
$ ls -R series
series:
Makefile
out
tmp

series/out:
0
1
2
3
4
5
6
7
8
9
total.pkl.gz

series/tmp:
$ dump-pickle total.pkl.gz ...
{'projects': {'t@': {'series': Date_Vector('2012-10-28', [ 2302.,  3086.,  3055.,  2950.,  2887.,  3078.,  3014.,  3209.,
        1185.], dtype=float32),
                     'total': 24766.0}}}
$ (ngrams-search --query 't@ candy' series/out)
t@ candy	32	2012-10-28	2012-11-04
$ (ngrams-search --detail --query 't@ candy' series/out)
2012-10-28	2	2302	868.81
2012-10-29	2	3086	648.088
2012-10-30	1	3055	327.332
2012-10-31	9	2950	3050.85
2012-11-01	10	2887	3463.8
2012-11-02	2	3078	649.773
2012-11-03	4	3014	1327.14
2012-11-04	2	3209	623.247
2012-11-05	0	1185	0.0
$ (ngrams-search --query 't@ sandy' series/out)
t@ sandy	119	2012-10-28	2012-11-04
$ (ngrams-search --detail --query 't@ sandy' series/out)
2012-10-28	8	2302	3475.24
2012-10-29	45	3086	14582.0
2012-10-30	42	3055	13748.0
2012-10-31	8	2950	2711.86
2012-11-01	4	2887	1385.52
2012-11-02	5	3078	1624.43
2012-11-03	4	3014	1327.14
2012-11-04	3	3209	934.871
2012-11-05	0	1185	0.0
$ (ngrams-search --query 't@ a' series/out)
t@ a	3222	2012-10-28	2012-11-05
$ (ngrams-search --detail --query 't@ taco' series/out)
2012-10-28	0	2302	0.0
2012-10-29	2	3086	648.088
2012-10-30	9	3055	2945.99
2012-10-31	3	2950	1016.95
2012-11-01	0	2887	0.0
2012-11-02	1	3078	324.886
2012-11-03	1	3014	331.785
2012-11-04	0	3209	0.0
2012-11-05	0	1185	0.0
$ ngrams-correlate ...
ngcor INFO     starting
ngcor INFO     setting up job
ngcor INFO     running job
ngcor INFO     cleaning up job
ngcor INFO     done
$ ls -R corr
corr:
Makefile
out
tmp

corr/out:
halloween:halloween.tsv
halloween:s%40ndy.tsv

corr/tmp:
$ head -10 corr/out/halloween:halloween.tsv
t@ happyhalloween	0.998131075392	2078.28	0.0
t@ happy halloween	0.996925606519	10847.5	0.0
t@ trick or	0.995727657015	3728.81	0.0
t@ thingsaboutme	0.993012003619	4067.8	0.0
t@ trick	0.990344975957	3728.81	0.0
t@ or treating	0.9881173924	2372.88	0.0
t@ nba	0.9881173924	4582.65	0.0
t@ treating	0.986329162782	2372.88	0.0
t@ halloween	0.984747458302	28135.6	623.247
t@ happy	0.983947838105	15593.2	5063.29
$ head -10 corr/out/halloween:s%40ndy.tsv
t@ super	0.99810059828	3600.65	1558.12
t@ sandy	0.99518702064	14582.0	934.871
t@ hurricane	0.993076652371	8425.15	0.0
t@ bonito	0.987933152078	2531.65	0.0
t@ car	0.985450405022	2945.99	1303.21
t@ storm	0.982598080446	2172.02	311.624
t@ nba nba	0.980498061745	3273.32	0.0
t@ nationaltextyourexday	0.980498001965	2291.33	0.0
t@ day	0.980068437556	13748.0	7962.84
t@ us	0.978356546112	6804.93	2531.65
